### PR TITLE
StackSets Takes StackTags and Retry Strategy When Throttling

### PR DESCRIPTION
**Pull request by xiwhuang**
*Issue #, if available:*

*Description of changes:*

* StackSet handlers take stack tags
* Add retry strategy for CloudFormation client in StackSet

*Testing*

* Manual create/update stack with stack tags.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/Configuration.java [hunk 1]
```diff
@@ -1,8 +1,28 @@
...
+        return new JSONObject(
+                new JSONTokener(this.getClass().getClassLoader().getResourceAsStream(schemaFilename)));
+    }
+
+    public Map<String, String> resourceDefinedTags(final ResourceModel resourceModel) {
+        if (CollectionUtils.isNullOrEmpty(resourceModel.getTags())) return null;
+
+        return resourceModel.getTags()
+                .stream()
+                .collect(Collectors.toMap(Tag::getKey, Tag::getValue, (value1, value2) -> value2));
```
Comment (ammokhov) [inline aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/Configuration.java:26]: dont think we should have this logic in stack sets, we can simply have this:
```java
        if(CollectionUtils.isNullOrEmpty(resourceModel.getTags()))
            return null;

        return resourceModel.getTags()
                .stream()
                .collect(Collectors.toMap(Tag::getKey, Tag::getValue));

```

File: aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/util/ClientBuilder.java [hunk 2]
```diff
@@ -18,8 +28,45 @@ public static CloudFormationClient getClient() {
...
+                        .retryPolicy(RetryPolicy.builder()
+                                .backoffStrategy(BackoffStrategy.defaultThrottlingStrategy())
+                                .throttlingBackoffStrategy(BackoffStrategy.defaultThrottlingStrategy())
+                                .numRetries(MAX_RETRIES)
+                                .retryCondition(OrRetryCondition.create(new RetryCondition[]{
+                                        RetryCondition.defaultRetryCondition(),
+                                        CloudFormationRetryCondition.create()
+                                }))
+                                .build())
+                        .build())
```
Comment (ammokhov) [inline aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/util/ClientBuilder.java:46]: have u tested it locally?

**Review by ammokhov — APPROVED** [review]
overall lgtm

File: aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/util/ClientBuilder.java (hunk 2 above)
Comment (xiwhuang) [reply aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/util/ClientBuilder.java:46]: There isn't really good way to test it. I added the unit test for CloudFormationRetryCondition as it is a customized condition.

**Comment by xiwhuang** [comment]
Tested to create 20 StackSets with 20 StackInstances at same time.

**Review by xaoyuli — APPROVED** [review]

**Pull request merged**